src/compat/array/differenceBy.ts

```diff
@@ -0,0 +1,63 @@
+import { differenceBy as differenceByToolkit } from '../../array/differenceBy.ts';
+import { flatten } from '../../array/flatten.ts';
+import { last } from '../../array/last.ts';
+import { isFunction } from '../../predicate/isFunction.ts';
+import { isArrayLikeObject } from '../predicate/isArrayLikeObject.ts';
+import { identity } from '../_internal/identity.ts';
+import { initial } from '../../array/initial.ts';
+
+/**
+ * Computes the difference between the initial array and multiple subsequent arrays after mapping their elements
+ * through a provided iteratee function or property key.
+ *
+ * This function takes an initial array, multiple arrays to compare against, and optionally, an iteratee function
+ * or property key. It returns a new array containing the elements from the initial array that are not present in
+ * the subsequent arrays, based on the identity calculated by the iteratee function or property key.
+ *
+ * If no iteratee function or property key is provided, the identity function is used by default. If the last
+ * argument is an array-like object, it is treated as the final array to compare against, and the identity function
+ * will be used as the default iteratee. If the last argument is an iteratee function or property key, that will be
+ * used to map the elements.
+ *
+ * @template T
+ * @param {T[]} arr - The initial array from which to derive the difference.
+ * @param {...T[][], T[] | ((value: T) => unknown) | keyof T} values - One or more arrays to compare against,
+ *   and optionally an iteratee function or property key. The last argument can be a function, a property key,
+ *   or an array (if the iteratee is not provided).
+ * @returns {T[]} A new array containing the elements from the initial array that do not have a corresponding
+ *   mapped identity in the subsequent arrays.
+ *
+ * @example
+ * const array1 = [{ id: 1 }, { id: 2 }, { id: 3 }];
+ * const array2 = [{ id: 2 }, { id: 4 }];
+ * const array3 = [{ id: 3 }];
+ * const result = differenceBy(array1, array2, array3, item => item.id);
+ * // result will be [{ id: 1 }]
+ *
+ * @example
+ * const array1 = [{ id: 1 }, { id: 2 }, { id: 3 }];
+ * const array2 = [{ id: 2 }, { id: 4 }];
+ * const result = differenceBy(array1, array2, 'id');
+ * // result will be [{ id: 1 }, { id: 3 }]
+ *
+ * @example
+ * const array1 = [1, 2, 3];
+ * const array2 = [2, 4];
+ * const array3 = [3];
+ * const result = differenceBy(array1, array2, array3);
+ * // result will be [1]
+ */
+export function differenceBy<T>(
```

Comment:
For types, we might refer to `@types/lodash`. 

`@types/lodash` provides several overloadings to `differenceBy`, could you reference it?

Reply:
@raon0211 
Thank you for the review. I plan to add function overloading based on `@types/lodash`.

However, I have one question to clarify my understanding. I initially thought that our compatibility functions should have the same types as `@types/lodash` to ensure 100% compatibility with Lodash.

However, I noticed that some functions, like `difference`, don't exactly match in type.

For example:
- `@types/lodash`: `difference<T>(array: List<T> | null | undefined, ...values: Array<List<T>>): T[];`
- `es-toolkit`: `function difference<T>(arr: readonly T[], ...values: Array<readonly T[]>): T[]`

This difference has me a bit confused. Could you clarify what needs to be the same and where it's acceptable to have differences?

Reply:
I think we should match the types from `@types/lodash` to ensure smooth migration. Maybe `readonly T[]` should be converted to `readonly T[] | ArrayLike<T> | null | undefined`.

Because of this change, we might have to wrap the arguments with `Array.from(...)` to convert them into an array..

Reply:
We initially aimed to keep changes to the original `es-toolkit`'s signature to a minimum, only referring to types from `@types/lodash` to address any type errors in the compatibility tests.

However, I'm wondering whether we should match our types more closely with those from `@types/lodash` to make the migration smoother.

For example, `readonly T[]` could be updated to `readonly T[] | ArrayLike<T>` (which is `List<T>`). If we go this route, we might need to wrap arguments with Array.from(...) to ensure they're converted to arrays where necessary.

Reply:
We recently launched a [Discord channel](https://discord.gg/vGXbVjP2nY) to help address these types of issues—feel free to join us! :)

Reply:
It would be great if I could work on this after we decide on the direction. I will join the Discord community. Thanks!

Reply:
@raon0211 

I updated the PR based on `@types/lodash`.  Please review this when you have some time.

Here’s a key point to note: While `@types/lodash` defines the `iteratee` parameter using `ValueIteratee<T>`, I changed it to `((value: T) => unknown) | PropertyKey` because i think these are the only valid cases supported by Lodash's `differenceBy` function. (`[PropertyName, any] | PartialShallow<T>` doesn't seem to be valid type)

For reference, here are the type definitions from @types/lodash:
```ts
type ValueIteratee<T> = ((value: T) => unknown) | IterateeShorthand<T>;

type IterateeShorthand<T> = PropertyKey | [PropertyKey, any] | PartialShallow<T>;

type PartialShallow<T> = {
  [P in keyof T]?: T[P] extends object ? Partial<T[P]> : T[P];
};

```